sklearn/gaussian_process/kernels.py

```diff
@@ -0,0 +1,940 @@
+    with different characteristic length-scales.
+
+    Only isotropic variant is supported at the moment.
+
+    Tunable kernel parameters
+    -------------------------
+    alpha : float > 0
+        Scale mixture parameter
+    l : float > 0
+        The length scale of the kernel.
+    """
+
+    def __init__(self, theta=[1.0, 1.0], thetaL=1e-5, thetaU=np.inf):
+        super(RationalQuadratic, self).__init__(theta, thetaL, thetaU)
+
+    @property
+    def theta(self):
+        return np.asarray([self.alpha, self.l])
+
+    @theta.setter
+    def theta(self, theta):
+        self.alpha = theta[0]
+        self.l = theta[1]
+
+    def __call__(self, X, Y=None, eval_gradient=False):
+        """ Return the kernel k(X, Y) and optionally its gradient.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples_X, n_features)
+            Left argument of the returned kernel k(X, Y)
+
+        Y : array, shape (n_samples_Y, n_features), (optional, default=None)
+            Right argument of the returned kernel k(X, Y). If None, k(X, X)
+            if evaluated instead.
+
+        eval_gradient : bool (optional, default=False)
+            Determines whether the gradient with respect to the kernel
+            hyperparameter is determined. Only supported when Y is None.
+
+        Returns
+        -------
+        K : array, shape (n_samples_X, n_samples_Y)
+            Kernel k(X, Y)
+
+        K_gradient : array (opt.), shape (n_samples_X, n_samples_X, n_dims)
+            The gradient of the kernel k(X, X) with repect to the
+            hyperparameter of the kernel. Only returned when eval_gradient
+            is True.
+        """
+        if Y is None:
+            dists = squareform(pdist(X, metric='sqeuclidean'))
+            tmp = dists / (2 * self.alpha * self.l ** 2)
+            base = (1 + tmp)
+            K = base ** -self.alpha
+            np.fill_diagonal(K, 1)
+        else:
+            if eval_gradient:
+                raise ValueError(
+                    "Gradient can only be evaluated when Y is None.")
+            dists = cdist(X, Y, metric='sqeuclidean')
+            K = (1 + dists / (2 * self.alpha * self.l ** 2)) ** -self.alpha
+
+        if eval_gradient:
+            K_gradient = np.empty((K.shape[0], K.shape[1], 2))
+            K_gradient[..., 0] = K * (-np.log(base) + tmp / base)
+            K_gradient[..., 1] = dists * K / (self.l ** 2 * base)
+            return K, K_gradient
+        else:
+            return K
+
+
+class ExpSineSquared(Kernel):
+    """ Exp-Sine-Squared kernel.
+
+    This kernel allows modelling periodic functions.
+
+    Only isotropic variant is supported at the moment.
+
+    Tunable kernel parameters
+    -------------------------
+    l : float > 0
+        The length scale of the kernel.
+    p : float > 0
+        The periodicity of the kernel.
+    """
+
+    def __init__(self, theta=[1.0, 1.0], thetaL=1e-5, thetaU=np.inf):
```

Comment:
remove list as default. Use parameter names as in docstring, i.e. `l` and `p` or more descriptive versions thereof.